torch/csrc/jit/passes/onnx/helper.cpp

```diff
@@ -96,5 +96,63 @@ Value* addInputToBlock(Block* block) {
   return block->addInput();
 }
 
+Node* CreateCastToBoolNode(Value* val, Graph* graph) {
+  Node* cast_node = graph->create(onnx::Cast);
+  cast_node->addInput(val);
+  cast_node->i_(attr::to, ONNX_TYPE_BOOL);
+  cast_node->output()->setType(BoolType::get());
+  return cast_node;
+}
+
+Node* InsertCastForCond(Value* cond_val, Graph* graph, Node* consumer_node) {
+  // prev:  cond_val -> consumer_node
+  // after: cond_val -> cast -> consumer_node
+  // NOTE: The cast is required because operators like PyTorch Greater/Less
+  //       return tensor in type torch.uint8. However the type for condition
+  //       input in ONNX Loop must be bool.
+  Node* cast_node = CreateCastToBoolNode(cond_val, graph);
+  cast_node->insertBefore(consumer_node);
+
+  consumer_node->replaceInputWith(cond_val, cast_node->output());
+  return cast_node;
+}
+
+bool IsCondCastRequired(Value* cond_val) {
+  const auto& type = cond_val->type();
+  if (auto tt = type->cast<TensorType>()) {
+    if (auto scalar_type = tt->scalarType()) {
+      return *scalar_type != c10::kBool;
+    }
+  }
+  return !type->isSubtypeOf(BoolType::get());
+}
+
+void updateLoopNodeInputs(Node* node) {
```

Comment:
why move the whole thing to helper? I think you can still change on api level to remove `fixup...` from `init.cpp` pass apis, and add it to `python_ir.cpp` as node api.